### DISCUSSION TITLE
Show rule giving an error

### DISCUSSION THREAD

**Discussion by psads-git** [typst/typst#2978] [Q&A] [answered]
With the code

```
#show "pytorch": `pytorch`

pytorch

```
I get the following error:

```
[2023-12-15T19:46:21Z DEBUG typst_preview::actor::editor] EditorActor: received message from mailbox: CompileStatus(CompileError)
error: maximum show rule depth exceeded
 = hint: check whether the show rule matches its own output
 = hint: this is a current compiler limitation that will be resolved in the future

```

Could someone please help me?

**Comment by PgBiel** [comment]
Your show rule is replacing "pytorch" with different text which also contains the word "pytorch", so your show rule is being applied to your show rule's output, repeating the process, which occurs a very large amount of times until the compiler gives up. In other words, you created a recursive show rule which matches its own output, as the error warns you about.

Instead, use

```typ
#show "pytorch": raw

pytorch
```

to wrap the word itself in a raw element, without creating another (otherwise identical) word. The rule won't recurse in that case.

**Reply by psads-git** [reply]
Thanks, @PgBiel. Now, the error is different:

```
[2023-12-15T20:02:38Z INFO  typst_ts_compiler::service::diag::console] /lixo.typ: Compilation failed after 3.926147ms
    
error: expected string, found content
    ┌─ /home/x/programas/r/pythonL/lixo.typ:235:17
    │
235 │ #show "pytorch": raw 
    │                  ^^^

help: error occurred while applying show rule to this text
    ┌─ /home/x/programas/r/pythonL/lixo.typ:237:0
    │
237 │ pytorch
    │ ^^^^^^^

```

**Reply by emilyyyylime** [reply]
Interesting; this seems to be currently impossible without some deep iterate-over-the-whole-document-and-split-and-reconnect-content shenanigans, unfortunately. One hack you can do is ``#show "pytorch": `pytorc\u{200B}h` ``.

The issue of recursive show rules is definitely a prevalent one among the Typst contributors, but the solution doesn't seem to be on the horizon just yet

**Reply by psads-git** [reply]
Thanks, @emilyyyylime. Unfortunately, the output is:

![image](https://github.com/typst/typst/assets/75945439/3672185c-a6e4-48a1-89ce-34db641c0cfe)

When I try:

```
#show "pytorch": x => [raw(#x)]

pytorch
```

The result is:

![image](https://github.com/typst/typst/assets/75945439/a8e76508-bc31-4f7c-bc0e-2228a66530f2)

**Reply by xkevio** [reply]
Workaround that works: `#show "pytorch": it => raw(it.text.at(0)) + raw(it.text.slice(1))`. Splits up the word so it won't recurse.

**Reply by psads-git** [reply, marked as answer]
Great, that works! Thanks, Kevin!
